File: server-mcp/internal/model/response/apikey.go
package response

import "time"

// APIKeyCreateResponse 创建 API Key 响应（仅创建时返回完整 key）
type APIKeyCreateResponse struct {
	ID          uint      `json:"id"`
	Name        string    `json:"name"`
	APIKey      string    `json:"api_key"`      // 完整 key，仅此一次显示
	TokenSuffix string    `json:"token_suffix"` // 后 4 位
	CreatedAt   time.Time `json:"created_at"`
}

// APIKeyListItem 列表项（脱敏显示）
type APIKeyListItem struct {
	ID          uint       `json:"id"`
	Name        string     `json:"name"`
	TokenSuffix string     `json:"token_suffix"` // 后 4 位
	LastUsedAt  *time.Time `json:"last_used_at"`
	CreatedAt   time.Time  `json:"created_at"`
}

File: web-mcp/src/api/apikey.ts
import service from '@/utils/request'
import type { ApiResponse } from '@/utils/request'

export interface APIKey {
  id: number
  name: string
  token_suffix: string
  last_used_at: string | null
  created_at: string
}

export interface APIKeyCreateRequest {
  name: string
}

export interface APIKeyCreateResponse {
  id: number
  name: string
  api_key: string  // 完整 key，仅创建时返回
  token_suffix: string
  created_at: string
}

// 获取 API Key 列表
export const getAPIKeys = (): Promise<APIKey[]> => {
  return service({
    url: '/api-keys/list',
    method: 'get'
  })
}

// 创建 API Key
export const createAPIKey = (data: APIKeyCreateRequest): Promise<APIKeyCreateResponse> => {
  return service({
    url: '/api-keys/create',
    method: 'post',
    data
  })
}

// 删除 API Key
export const deleteAPIKey = (id: number): Promise<null> => {
  return service({
    url: `/api-keys/${id}`,
    method: 'delete'
  })
}
